### PR TITLE
Add AppVeyor for Windows CI

### PR DESCRIPTION
**Pull request by vodik**
In light of a few recent Windows issues, maybe it makes sense to leverage AppVeyor and get a Windows CI build going.

Its already catching some issues.

### CODE REVIEW

**Comment by vodik** [comment]
This should be stable: https://ci.appveyor.com/project/vodik/hy/build/1.0.8

Once merged I can turn it on for hylang/hy

**Comment by Kodiologist** [comment]
Can you turn off the automatic AppVeyor check for every new PR?

**Comment by vodik** [comment]
I set it to skip branches without `appveyor.yml` present, so its on but shouldn't spam every PR anymore until this gets merged.

**Comment by vodik** [comment]
@Kodiologist style mostly. IMHO format strings should be treated as constants as much as possible - though I guess it would be safe in this case.

File: hy/contrib/hy_repr.hy [hunk 1]
```diff
@@ -10,6 +10,13 @@
   [hy._compat [PY3 PY36 str-type bytes-type long-type]]
   [hy.models [HyObject HyExpression HySymbol HyKeyword HyInteger HyFloat HyComplex HyList HyDict HySet HyString HyBytes]])
 
+(defmacro strftime [time formatstr]
+  "Version of strftime that handles some Windows/POSIX portability"
+  (import platform)
+  (when (= (platform.system) "Windows")
+    (setv formatstr (.replace (str formatstr) "%-" "%#")))
```
Comment (vodik) [inline hy/contrib/hy_repr.hy]: Slightly annoying that `HyObject.replace` collides with `str.replace`.

**Comment by Kodiologist** [comment]
> @Kodiologist style mostly. IMHO format strings should be treated as constants as much as possible - though I guess it would be safe in this case.

Wouldn't it suffice to write something like

    (.replace (.strftime x "%Y %m %d %H %M %S") "  " " ")

? Then the format string is a constant, and you don't need to check the platform.

**Comment by vodik** [comment]
@Kodiologist Is the `strftime` macro acceptable? Similar idea but we shunt it to compile time.

**Comment by Kodiologist** [comment]
I don't see how that would work. The line I wrote manipulates the return value of `.strftime`, which doesn't exist till runtime.

**Comment by Kodiologist** [comment]
I'm closing this for lack of response.

**Pull request closed without merging**